### ISSUE TITLE
Faster loading of the main VP screen

### ISSUE THREAD

**Issue by borekb** [versionpress/versionpress#1152]
While VersionPress' UI is generally fast thanks to being written in JavaScript / React, the initial loading is slower than most other sections like posts or plugins. It's an acceptable "tax" during the first load but it's bothering me if I frequently switch between e.g. posts and VersionPress. The better experience would be to have the table cached locally and only load the fresh data asynchronously, in the background.

**Comment by vasek17** [comment]
Let me understand this:
1. When I open VersionPress page, old data is loaded.
2. After a few seconds new data is loaded, saved to local cache and a notification (#427) is displayed.
3. At this point, either refresh via notification or refreshing the page in browser will display new data.
Is it correct?

What about search?
Let's assume I have an active search query, then close the tab and open it again. What data should I see? The most recently loaded data (filtered) or the most recently loaded **unfiltered** data?

I think for this case better solution would be to bundle the data in the initial server response.

**Comment by borekb** [comment]
The experience should ideally be as if the whole wp-admin was stateful SPA app, i.e., when I switch away from the VP page and then go back again, I see the exact same thing immediately. How it's best realized I'm not sure.

**Comment by vasek17** [comment]
Well, the rest of wp-admin doesn't work this way. When I switch away from any page and then go back, I see the most recent data (which may not be the same thing as when leaved the page).
I would prefer being consistent with WP in this behaviour.

**Comment by borekb** [comment]
I agree but the current experience is just slow. I'd rather see outdated data for a short while than to wait hundreds of milliseconds on every visit to the main VP page.

**Comment by vasek17** [comment]
I agree that it is slow. 
But displaying old data is visually faster but confusing in most cases (it is quite likely that I will do some action logged to the VP table when I go away from VP page). And for the actual usage I need the current data so it would not be much faster.
What about bundling the data in the initial request?

**Comment by borekb** [comment]
Bundling is probably the best approach. Is that easy enough to do?

**Comment by vasek17** [comment]
I'll look into the server side. Maybe it would require some refactoring of API endpoints to be directly callable. 
The frontend part will be almost the same as in the local caching way (only restore from a global variable instead of localStorage).

**Comment by vasek17** [comment]
This approach wouldn't work for pages or active search query in the future (they both use hash fragment in URL). The URL hash fragments are not sent to the server so it is not possible to bundle the correct data.

So it would be faster when you'll open VersionPress from wp-admin menu or using some link but not when e.g. refreshing the page or using browser Undo button after switching away from VP page.
